ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -613,14 +611,7 @@ private void performHardDeletes(List<MessageInfo> messageInfoList) throws StoreE
         }
         logWriteInfo.logSegment.writeFrom(logWriteInfo.channel, logWriteInfo.offset, logWriteInfo.size);
         metrics.hardDeleteDoneCount.inc(1);
-        throttler.maybeThrottle(logWriteInfo.size);
-      }
-    } catch (InterruptedException e) {
-      if (enabled.get()) {
-        // We throw here because we do not want the tokens to be updated.
-        throw new StoreException("Got interrupted during hard deletes", StoreErrorCodes.Unknown_Error);
-      } else {
-        throw new StoreException("Got interrupted as store is shutting down", StoreErrorCodes.Store_Shutting_Down);
+        diskIOScheduler.getSlice(DiskManager.CLEANUP_OPS_JOB_NAME, DiskManager.CLEANUP_OPS_JOB_NAME, logWriteInfo.size);
```

Comment:
I feel that the jobType/jobId should be passed in an arg to this class during initialization. In general, anyone (class) who is going to use DiskIOScheduler should be passed in a jobType/jobId as well. Or another option is that DiskManager can expose a static method called getCleanUpJobName(). So that we don't restrict one class to use only one type of throttler/jobType.